scripts/build_osm_network.py

```diff
@@ -613,7 +613,7 @@ def fix_overpassing_lines(lines, buses, distance_crs, tol=1):
         for point in nearest_points_list:
             # Split the line at the current point
             # The split function returns a GeometryCollection, so we need to convert it to a list
-            split_lines = split(split_line[-1], point)
+            split_lines = split(snap(split_line[-1], point, 1e-9), point)
```

Comment:
Could it be a good idea to add a short comment to explain the meaning of `1e-9`?